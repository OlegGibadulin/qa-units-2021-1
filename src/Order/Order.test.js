import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import {fakeOrders} from '../data/fakeOrders'
import toJson from "enzyme-to-json";

import Order from './Order';

configure({adapter: new Adapter()});

import getDate from '../utils/getDate'
jest.mock(getDate)

describe('Order.js', () => {
  it('empty props', () => {
    const wrapper = shallow(<Order/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('shop and date are null', () => {
    const wrapper = shallow(<Order order={{shop: null, date: null}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('shop and date are null', () => {
    const wrapper = shallow(<Order order={{shop: null, date: null}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render without items', () => {
    const wrapper = shallow(<Order order={{shop: 'a', date: new Date('December 1, 2020 00:00:00')}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with items', () => {
    const wrapper = shallow(<Order order={{shop: 'a', date: new Date('December 1, 2020 00:00:00'), items: ['item1','item2']}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('check getDate', () => {
    const wrapper = shallow(<Order order={fakeOrders} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
