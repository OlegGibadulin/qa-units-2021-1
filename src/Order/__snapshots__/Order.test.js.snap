// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js check getDate 1`] = `""`;

exports[`Order.js empty props 1`] = `""`;

exports[`Order.js render with items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      a
    </span>
    <span
      id="date"
    />
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      item1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      item2
    </div>
  </div>
</div>
`;

exports[`Order.js render without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      a
    </span>
    <span
      id="date"
    />
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js shop and date are null 1`] = `""`;

exports[`Order.js shop and date are null 2`] = `""`;
